packages/workflow/test/ExpressionExtensions/StringExtensions.test.ts

```diff
@@ -15,28 +14,23 @@ describe('Data Transformation Functions', () => {
 			expect(evaluate('={{"".isEmpty()}}')).toEqual(true);
 		});
 
-		test('.hash() should work correctly on a string', () => {
-			expect(evaluate('={{ "12345".hash("sha256") }}')).toEqual(
-				stringExtensions.functions.hash('12345', ['sha256']),
-			);
-
-			expect(evaluate('={{ "12345".hash("sha256") }}')).not.toEqual(
-				stringExtensions.functions.hash('12345', ['MD5']),
-			);
-
-			expect(evaluate('={{ "12345".hash("MD5") }}')).toEqual(
-				stringExtensions.functions.hash('12345', ['MD5']),
-			);
-
-			expect(evaluate('={{ "12345".hash("sha256") }}')).toEqual(
-				'5994471abb01112afcc18159f6cc74b4f511b99806da59b3caf5a9c173cacfc5',
-			);
-		});
-
-		test('.hash() alias should work correctly on a string', () => {
-			expect(evaluate('={{ "12345".hash("sha256") }}')).toEqual(
-				'5994471abb01112afcc18159f6cc74b4f511b99806da59b3caf5a9c173cacfc5',
-			);
+		describe('.hash()', () => {
+			const hash = (algo: string) => createHash(algo).update('12345').digest('hex');
+			const tests = [
+				['md5', hash('md5')],
+				['sha1', hash('sha1')],
+				['sha224', hash('sha224')],
+				['sha256', hash('sha256')],
+				['sha384', hash('sha384')],
+				['sha512', hash('sha512')],
+				['sha3', hash('sha3-512')],
+			] as const;
+			for (const [hashFn, hashValue] of tests) {
+				test(`should work for ${hashFn}`, () => {
+					expect(evaluate(`={{ "12345".hash("${hashFn}") }}`)).toEqual(hashValue);
+					expect(evaluate(`={{ "12345".hash("${hashFn.toLowerCase()}") }}`)).toEqual(hashValue);
+				});
+			}
```

Comment:
Somewhat matter of preference which way to use, but there's also `test.each` that creates a test case with multiple sets of test data. IMHO makes it bit clearer:

```ts
test.each([
				['md5', hash('md5')],
				['sha1', hash('sha1')],
				['sha224', hash('sha224')],
				['sha256', hash('sha256')],
				['sha384', hash('sha384')],
				['sha512', hash('sha512')],
				['sha3', hash('sha3-512')],
])("should work for %", (hashFn, hashValue) => {
  ...
})
```

Reply:
Also, if we plan to change the crypto lib in the future as well, would be nice to have a test case with each algorithm that uses `.toMatchInlineSnapshot` to make sure we don't introduce any changes in our code or in dependencies that would change the result of the hashing

Reply:
Thanks. TIL about `test.each`.
regarding snapshot testing: I've never done that outside frontend tests.
Maybe I could instead replace the `hash` calls with static strings?